Detectors/Base/include/DetectorsBase/GeometryManager.h

```diff
@@ -73,6 +101,99 @@ class GeometryManager : public TObject
 
   ClassDefOverride(GeometryManager, 0); // Manager of geometry information for alignment
 };
+
+
+//_____________________________________________________________________________________
+template <typename T> 
+GeometryManager::MatBudget GeometryManager::MeanMaterialBudget(const Point3D<T> &start, const Point3D<T> &end)
+{
+  // 
+  // Calculate mean material budget and material properties between 
+  //    the points "start" and "end".
+  //
+  // "mparam" - parameters used for the energy and multiple scattering
+  //  corrections: 
+  //
+  // see MatBudget data members for provided information
+  //
+  //  Origin:  Marian Ivanov, Marian.Ivanov@cern.ch
+  //
+  //  Corrections and improvements by
+  //        Andrea Dainese, Andrea.Dainese@lnl.infn.it,
+  //        Andrei Gheata,  Andrei.Gheata@cern.ch
+  //
+  //  Ported to O2: ruben.shahoyan@cern.ch
+  //
+  
+  double length, startD[3] = {start.X(),start.Y(),start.Z()};
+  double dir[3] = {end.X()-start.X(), end.Y()-start.Y(), end.Z()-start.Z()};
+  if ( (length = dir[0]*dir[0] + dir[1]*dir[1] + dir[2]*dir[2]) < TGeoShape::Tolerance()*TGeoShape::Tolerance() ) {
+    return MatBudget();  // return empty struct
+  }
+  length = TMath::Sqrt(length);
+  double invlen = 1./length;
+  for (int i=3;i--;) dir[i] *= invlen;
+
+  // Initialize start point and direction
+  TGeoNode *currentnode = gGeoManager->InitTrack(startD, dir);
+  if (!currentnode) {
+    LOG(ERROR) << "start point out of geometry: " << start << FairLogger::endl;
+    return  MatBudget();  // return empty struct
+  }
+
+  MatBudget budTotal,budStep;  
+  budStep.accountMaterial( currentnode->GetVolume()->GetMedium()->GetMaterial() );
+  budStep.length = length;
+    
+  // Locate next boundary within length without computing safety.
+  // Propagate either with length (if no boundary found) or just cross boundary
+  gGeoManager->FindNextBoundaryAndStep(length, kFALSE);
+  Double_t stepTot = 0.0; // Step made
+  Double_t step = gGeoManager->GetStep();
+  // If no boundary within proposed length, return current step data
+  if (!gGeoManager->IsOnBoundary()) {
+    budStep.meanX2X0 = budStep.length / budStep.meanX2X0;
+    return MatBudget(budStep);
+  }
+  // Try to cross the boundary and see what is next
+  Int_t nzero = 0;
+  while (length>TGeoShape::Tolerance()) {
+    if (step<2.*TGeoShape::Tolerance()) nzero++;
+    else nzero = 0;
```

Comment:
consider putting systematically `{}` even around single line bodies according to coding conventions.